core/src/main/scala/cats/free/FreeSelective.scala

```diff
@@ -0,0 +1,146 @@
+package cats.free
+
+import cats.arrow.FunctionK
+import cats.{Applicative, Bifunctor, Functor, Selective, ~>}
+import cats.selected.instances.option.catsStdSelectiveForOption
+import cats.instances.either.catsStdBitraverseForEither
+import cats.syntax.either._
+
+/**
+  * data Select f a where
+  *   Pure :: a -> Select f a
+  *   Select :: Select f (Either a b) -> f (a -> b) -> Select f b
+  *
+  * Free rigid selective functors. Rigidity is enforced
+  * as the definition is based on the normal form that can
+  * only be found for rigid selective functors.
+  * This is to say that not all selective functors are free.
+  * (A good thing, in most cases).
+  */
+sealed trait FreeSelective[F[_], A] {
+  type Source
+}
+object FreeSelective {
+  abstract class Pure[F[_], A] extends FreeSelective[F, A] {
+    type Source = A
+    val run: A
+  }
+  object Pure {
+    def apply[F[_], A](x: A): Pure[F, A] = new Pure[F, A] {
+      val run: A = x
+    }
+    def unapply[F[_], A](arg: Pure[F, A]): Option[A] = Option(arg.run)
+    type Aux[F[_], A, B] = Pure[F, A] {
+      type Source = B
+    }
+  }
```

Comment:
Honestly, should probably delete this stuff... forgot I was playing with trying to get a nice unapply interface here.

Reply:
Fixed now.